cluster/store.go

```diff
@@ -599,7 +599,7 @@ func (st *Store) openDatabase(ctx context.Context) {
 // the leader may decide to send a snapshot. Consequently, the follower must update its state accordingly.
 func (st *Store) reloadDBFromSnapshot() (success bool) {
 	defer func() {
-		if success {
+		if success && !st.cfg.MetadataOnlyVoters {
```

Comment:
I wanted to have it safe and check twice to avoid someone later down the line calling the function without realising. Also `reloadDBFromSnapshot` is also called on `Restore` for RAFT snapshot and it doesn't go through `reloadDBFromSchema` first :)